src/zenml/integrations/seldon/services/seldon_deployment.py

```diff
@@ -0,0 +1,279 @@
+            zenml_pipeline_step_name=self.config.pipeline_step_name or "",
+            zenml_service_uuid=str(self.uuid),
+        )
+
+    def provision(self) -> None:
+        """Provision or update the remote Seldon Core deployment instance to
+        match the current configuration.
+        """
+        client = self._get_client()
+
+        name = self.seldon_deployment_name
+        deployment = SeldonDeployment.build(
+            name=name,
+            model_uri=self.config.model_uri,
+            model_name=self.config.model_name,
+            implementation=self.config.implementation,
+            secret_name=self.config.secret_name,
+            labels=self._get_seldon_deployment_labels(),
+        )
+        deployment.spec.replicas = self.config.replicas
+        deployment.spec.predictors[0].replicas = self.config.replicas
+
+        # check if the Seldon deployment already exists
+        try:
+            client.get_deployment(name=name)
+            # update the existing deployment
+            client.update_deployment(deployment)
+        except SeldonDeploymentNotFoundError:
+            # create the deployment
+            client.create_deployment(deployment=deployment)
+
+    def deprovision(self, force: bool = False) -> None:
+        """Deprovision the remote Seldon Core deployment instance.
+
+        Args:
+            force: if True, the remote deployment instance will be
+                forcefully deprovisioned.
+        """
+        client = self._get_client()
+        name = self.seldon_deployment_name
+        try:
+            client.delete_deployment(name=name, force=force)
+        except SeldonDeploymentNotFoundError:
+            pass
+
+    @property
+    def prediction_url(self) -> Optional[str]:
+        """The prediction URI exposed by the prediction service.
+
+        Returns:
+            The prediction URI exposed by the prediction service, or None if
+            the service is not yet ready.
+        """
+        if not self.is_running:
+            return None
+        # the namespace is either explicitly configured or implicitly
+        # determined by the in-cluster Kuberenetes configuration
+        namespace = self.config.namespace or self._get_client().namespace
+        if not namespace:
+            # shouldn't happen if the service is running, but we need to
+            # appease the mypy type checker
+            return None
+        return os.path.join(
+            self.config.base_url,
+            "seldon",
+            namespace,
+            self.seldon_deployment_name,
+            "api/v0.1/predictions",
+        )
+
+    def predict(self, request: "NDArray[Any]") -> "NDArray[Any]":
+        """Make a prediction using the service.
+
+        Args:
+            request: a numpy array representing the request
+
+        Returns:
+            A numpy array representing the prediction returned by the service.
+        """
```

Comment:
Does this mean we will only support prediction models that take numpy arrays as input?

Reply:
No. The Seldon Core prediction API [supports several data types](https://docs.seldon.io/projects/seldon-core/en/latest/reference/apis/external-prediction.html#rest-api). We only offer this predict function as a convenience that can be used to contact the deployment service and run a quick inference request. Implementing a full prediction client is out of scope for this integration. Besides, there is an entire seldon-core Python library dedicated to this.